### PR TITLE
introduce dvc.io module to consolidate io utilities

### PR DESCRIPTION
**Pull request by skshetry**
Related to https://github.com/iterative/dvc/issues/5347

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

**Comment by efiop** [comment]
Maybe `io` -> `ui`? `io` has lowlevel connotation, at least to me. `ui` is logically about printing, tables, etc.

**Comment by pmrowla** [comment]
> Maybe `io` -> `ui`? `io` has lowlevel connotation, at least to me. `ui` is logically about printing, tables, etc.

Same for me, especially in the python sense, `io` as a module sounds more like something for directly manipulating streams (same as python stdlib `io`)

File: dvc/io/__init__.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            "warn": {"color": "yellow"},
+            "error": {"color": "red", "style": "bold"},
+        }
+        self.theme = defaultdict(lambda: defaults or {}, theme)
+
+    def format(self, message: str, style: str = None, **kwargs) -> str:
+        return colorize(message, **self.theme[style])
+
+
+class IO:
```
Comment (pmrowla) [inline dvc/io/__init__.py:27]: I'm still in favor of naming this something besides `IO`, whether it's `UI` or `Console` or something else entirely.

`IO` to me still makes it look like this is supposed to be related to `io.IOBase` or `io.TextIOBase` for reading/writing text streams. We do have a `write` method, but it isn't really an arbitrary stream write, it's a formatted wrapper for `print`.

File: dvc/io/__init__.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        self.disabled: bool = disable
+
+    def success(self, message: str) -> None:
+        self.write(message, style="success")
+
+    def error(self, message: str) -> None:
+        self.error_write(message, style="error")
+
+    def warn(self, message: str) -> None:
+        self.error_write(message, style="warn")
```
Comment (pmrowla) [inline dvc/io/__init__.py:49]: Is the idea here that we will eventually replace the use of `logging` with these methods?

Or are these just for console UI output messages which would be separate from internal logging calls?

Comment (skshetry) [reply dvc/io/__init__.py]: If we start printing in colors, they will come in handy.

**Comment by skshetry** [comment]
@pmrowla @pared @efiop  @isidentical, guys, PTAL again. I changed the module name to `dvc.ui` and the class to `Console` (as I plan to add more `ui` related components to the module, though you'll probably use it as `ui.success` etc.).

**Comment by skshetry** [comment]
I planned to add `rich_table` support in this same PR, but looks like it'll take some time to unify it, especially the styling (and/or adding markdown/csv/json formats into it). So, I'd like to merge this and work on `Table` on a different PR.

**Review by pared — APPROVED** [review]

**Review by efiop — APPROVED** [review]
Great start! :fire:

**Pull request merged**